gnd/src/main/java/com/google/android/gnd/ui/offlinebasemap/viewer/OfflineBaseMapViewerViewModel.java

```diff
@@ -101,9 +101,20 @@ private double tileSourceStorageSize(TileSource tileSource) {
   /**
    * Removes the offline area associated with this viewmodel from the device by removing all tile
    * sources that are not included in other areas and removing the area from the db.
+   *
+   * <p>Terminates the upstream click processor so that the resulting stream completes.
    */
-  public Completable onRemoveClick() {
-    return offlineBaseMapRepository.deleteArea(this.offlineArea.getValue().getId());
+  @Hot(terminates = true)
```

Comment:
@gino-m This is the `Hot` terminating case I mentioned in our last standup--I was slightly inaccurate, as the reason the click stream *must* complete is because we `flatMap` into a Completable which will *only* complete granted both the returned completable *and* its upstream complete.

I think it makes sense to model this as a completable, since we're ultimately calling a completable DB operation, but doing so forces us to call `complete` on the click processor to in turn propagate the `complete` to the completable.

The old way this was written (more imperative style) also works; but anecdotally I've found that this approach actually makes the UI run faster (I guess because the call no longer blocks on the UI thread VS the some amount of blocking imposed by the original imperative solution?)